src/jit/lsraarmarch.cpp

```diff
@@ -403,13 +403,22 @@ void Lowering::TreeNodeInfoInitPutArgReg(
     info.srcCount++;
 
     // Set the register requirements for the node.
-    const regMaskTP argMask = genRegMask(argReg);
+    regMaskTP argMask = genRegMask(argReg);
+#ifdef ARM_SOFTFP
+    // If type of node is `long` then it is actually `double`.
+    // The actual `long` types must have been transformed as a field list with two fields.
+    if (node->TypeGet() == TYP_LONG)
+    {
+        info.srcCount++;
+        argMask |= genRegMask(REG_NEXT(argReg));
+    }
+#endif // ARM_SOFTFP
     node->gtLsraInfo.setDstCandidates(m_lsra, argMask);
     node->gtLsraInfo.setSrcCandidates(m_lsra, argMask);
 
     // To avoid redundant moves, have the argument operand computed in the
     // register in which the argument is passed to the call.
-    node->gtOp.gtOp1->gtLsraInfo.setSrcCandidates(m_lsra, m_lsra->getUseCandidates(node));
+    node->gtOp.gtOp1->gtLsraInfo.setDstCandidates(m_lsra, m_lsra->getUseCandidates(node));
```

Comment:
Yes. This is a somewhat tricky case that I don't think we've handled before - that is, where we are consuming sources from a node that produces multiple registers, and where they have fixed reg requirements. This would need to be handled near line 4013 in your version of lsra.cpp, where it sets `fixedAssignment` assuming that there will only be a single register in that case.